### PR TITLE
Remove bower

### PR DESCRIPTION
**Pull request by simison**
Removes Bower package-manager dependency by installing everything via `npm` and requiring scripts directly from `node_modules`.

Todo:

Do something about `/lib/*` assets (mainly [image files](https://github.com/Trustroots/trustroots/blob/d1695a416294854baf7d0dc528bd5131f73e83a5/modules/core/client/less/modules/chosen.less#L13) referenced at core-module and [JS files for Firebase](https://github.com/Trustroots/trustroots/blob/f4bd051562b112e28b4bbeb920374b7a7713dbb2/public/push-messaging-sw.js#L6-L7)) in a few JS files and CSS files.

We could either copy these over to `public` folder with Gulp or use express static-serve handle them.

#### Testing Instructions

- Start the app both in dev (`npm start`) and prod (`NODE_ENV=production npm start`) modes
- Click around and see if you see 404s in console or anything off/missing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,20 +52,40 @@
...
+    "angular": "1.5.11",
+    "angulargrid": "~0.6.0",
+    "angulartics-google-analytics": "~0.5.0",
+    "angulartics": "~1.6.0",
     "async": "~2.6.1",
     "autolinker": "~1.7.1",
     "autoprefixer": "~9.1.5",
     "basic-auth": "~2.0.1",
     "body-parser": "~1.18.3",
-    "bower": "~1.8.4",
```
Comment (nicksellen) [inline package.json]: Bower is still used inside `message-center` lib in the postinstall script [1], and install fails if you don't have it (I guess you have bower globally installed?)

[1] https://github.com/e0ipso/message-center#

File: config/assets/default.js [hunk 2]
```diff
@@ -14,160 +14,160 @@ module.exports = {
...
+        'node_modules/leaflet/dist/leaflet-src.js',
+        'node_modules/angular-simple-logger/dist/angular-simple-logger.js', // Required by angular-leaflet-directive
+        'node_modules/PruneCluster/dist/PruneCluster.js',
+        'node_modules/ui-leaflet/dist/ui-leaflet.js',
+        'node_modules/leaflet-active-area/src/leaflet.activearea.js',
+        'node_modules/angular-waypoints/dist/angular-waypoints.all.js',
+        'node_modules/ng-file-upload/dist/ng-file-upload.js',
+        'node_modules/message-center/message-center.js',
+        'node_modules/chosen/chosen.jquery.js',
+        'node_modules/angular-chosen-localytics/dist/angular-chosen.js',
```
Comment (nicksellen) [inline config/assets/default.js]: I'm seeing an error coming from this lib when on http://localhost:3000/profile/edit.

> Error: element.chosen is not a function
> init/<@http://localhost:3000/node_modules/angular-chosen-localytics/dist/angular-chosen.js?c=f4bd0515:94:31

Comment (nicksellen) [reply config/assets/default.js]: (and do not get that error on master branch)

File: config/assets/default.js [hunk 3]
```diff
@@ -14,160 +14,160 @@ module.exports = {
...
+        'node_modules/angular-ui-bootstrap/src/stackedMap/stackedMap.js',
+        'node_modules/angular-ui-bootstrap/src/tabs/tabs.js',
+        'node_modules/angular-ui-bootstrap/src/tooltip/tooltip.js',
+        'node_modules/angular-ui-bootstrap/src/typeahead/typeahead.js',
+        'node_modules/moment/moment.js',
+        'node_modules/angular-moment/angular-moment.js',
+        'node_modules/medium-editor/dist/js/medium-editor.js',
+        'node_modules/leaflet/dist/leaflet-src.js',
+        'node_modules/angular-simple-logger/dist/angular-simple-logger.js', // Required by angular-leaflet-directive
+        'node_modules/PruneCluster/dist/PruneCluster.js',
```
Comment (nicksellen) [inline config/assets/default.js]: Getting an error on the map page.

> Error: PruneClusterForLeaflet is not defined
> SearchMapController@http://localhost:3000/modules/search/controllers/search-map.client.controller.js?c=f4bd0515:22:5

File: package.json (hunk 1 above)
Comment (simison) [reply package.json]: Good catch!

We should probably just include `message-center` directly in Trustroots codebase; it's unmaintained anyway.

**Comment by nicksellen** [comment]
Closing in favour of https://github.com/Trustroots/trustroots/pull/784 which is built on top.

**Pull request closed without merging**